src/signalfx_metadata.py

```diff
@@ -0,0 +1,255 @@
+        putnotif(property_name, property_value)
+
+
+# source http://code.activestate.com/recipes/439093-get-names-of-all-up-network-interfaces-linux-only/#c7
+# trying not to bundle too much code along with this but this was the best, most portable version of what
+# i needed
+def all_interfaces():
+    is_64bits = sys.maxsize > 2 ** 32
+    struct_size = 40 if is_64bits else 32
+    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    max_possible = 8  # initial value
+    while True:
+        bytes = max_possible * struct_size
+        names = array.array('B', '\0' * bytes)
+        outbytes = struct.unpack('iL', fcntl.ioctl(
+            s.fileno(),
+            0x8912,  # SIOCGIFCONF
+            struct.pack('iL', bytes, names.buffer_info()[0])
+        ))[0]
+        if outbytes == bytes:
+            max_possible *= 2
+        else:
+            break
+    namestr = names.tostring()
+    return [(namestr[i:i + 16].split('\0', 1)[0],
+             socket.inet_ntoa(namestr[i + 20:i + 24]))
+            for i in range(0, outbytes, struct_size)]
+
+
+def get_interfaces(host_info={}):
+    for interface, ipaddress in all_interfaces():
+        if ipaddress == "127.0.0.1":
+            continue
+        host_info["ipaddress_" + interface] = ipaddress
+        host_info["fqdn_" + interface] = socket.gethostbyaddr(ipaddress)[0]
+    return host_info
+
+
+def get_cpu_info(host_info={}):
+    info_raw = popen(["cat", "/proc/cpuinfo"])
+    model = ""
+    nb_cpu = 0
+    nb_cores = 0
+    nb_units = 0
+    for p in string.split(info_raw, "\n"):
+        if ':' in p:
+            x, y = map(lambda x: string.strip(x), string.split(p, ':', 1))
+            if x.startswith("physical id"):
+                if nb_cpu < int(y):
+                    nb_cpu = int(y)
+            if x.startswith("cpu cores"):
+                if nb_cores < int(y):
+                    nb_cores = int(y)
+            if x.startswith("processor"):
+                if nb_units < int(y):
+                    nb_units = int(y)
+            if x.startswith("model name"):
+                model = y
+
+    nb_cpu += 1
+    nb_units += 1
+    host_info["cpu_model"] = model
+    host_info["physical_cpus"] = str(nb_cpu)
+    host_info["cpu_cores"] = str(nb_cores)
+    host_info["logical_cpus"] = str(nb_units)
+    return host_info
+
+
+# platform module throws exceptions while running within the python collectd plugin
+# shell out to uname instead
+def get_kernel_info(host_info={}):
+    host_info["kernel_name"] = popen(["uname", "-s"])
+    host_info["kernel_release"] = popen(["uname", "-r"])
+    host_info["kernel_version"] = popen(["uname", "-v"])
+    host_info["machine"] = popen(["uname", "-m"])
+    host_info["processor"] = popen(["uname", "-p"])
+    host_info["operating_system"] = popen(["uname", "-o"])
+    return host_info
+
+
+def get_aws(thing, host_info={}):
+    try:
+        r = requests.get("http://169.254.169.254/latest/meta-data/%s" % thing, timeout=0.1)
```

Comment:
Requests is not in the python stdlib. Make sure you document the need for a pip install.